pkg/liqonet/ipam/ipam_integration_test.go

```diff
@@ -0,0 +1,165 @@
+package ipam_test
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/containernetworking/plugins/pkg/ns"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/kubernetes/pkg/util/slice"
+
+	"github.com/liqotech/liqo/pkg/consts"
+	"github.com/liqotech/liqo/pkg/liqonet/errors"
+	liqonetIpam "github.com/liqotech/liqo/pkg/liqonet/ipam"
+)
+
+var _ = Describe("IPAM integration tests", func() {
+	Describe("Endpoint IP mapping", func() {
+		Context("Ask for a mapping into a non-existing cluster", func() {
+			It("IPAM module should return an error in gRPC call", func() {
+				_, err := ipamInt.MapEndpointIP(context.Background(), &liqonetIpam.MapRequest{
+					ClusterID: clusterID2,
+					Ip:        localEndpointIP,
+				})
+				Expect(err.Error()).To(ContainSubstring(fmt.Sprintf("cluster %s has not a network configuration", clusterID2)))
+			})
+		})
+		Context("Passing an empty clusterID", func() {
+			It("IPAM module should return an error in gRPC call", func() {
+				_, err := ipamInt.MapEndpointIP(context.Background(), &liqonetIpam.MapRequest{
+					ClusterID: "",
+					Ip:        "10.0.0.2",
+				})
+				Expect(err.Error()).To(ContainSubstring(fmt.Sprintf("%s must be %s", consts.ClusterIDLabelName, errors.StringNotEmpty)))
+			})
+		})
+		Context("Ask for a mapping of an invalid IP", func() {
+			It("IPAM module should return an error in gRPC call", func() {
+				_, err = ipamInt.MapEndpointIP(context.Background(), &liqonetIpam.MapRequest{
+					ClusterID: clusterID4,
+					Ip:        "10.0.2",
+				})
+				Expect(err.Error()).To(ContainSubstring("cannot parse endpointIP 10.0.2"))
+			})
+		})
+		Context("Ask to map the IP of a local endpoint", func() {
+			It("IPAM module should return the input IP mapped on Local NAT PodCIDR network", func() {
+				response, err := ipamInt.MapEndpointIP(context.Background(), &liqonetIpam.MapRequest{
+					ClusterID: clusterID4,
+					Ip:        localEndpointIP,
+				})
+				Expect(err).To(BeNil())
+				Expect(response.GetIp()).To(Equal("10.70.0.20")) // Local NAT PodCIDR is 10.70.0.0/24
+			})
```

Comment:
Those first assertions (lines 18-54) look like a unit test for the IPAM service validating the correct input, I don't know if they deserve to be put in the corresponding unit tests.